SharedGraphics/Painter.cpp

```diff
@@ -204,6 +204,36 @@ void Painter::draw_bitmap(const Point& p, const GlyphBitmap& bitmap, Color color
     }
 }
 
+void Painter::blit_scaled(const Point& position, const GraphicsBitmap& source, const Rect& src_rect, float hscale, float vscale)
```

Comment:
This API is super awkward. It should instead take the `dst_rect` as an input parameter, and compute the `hscale` and `vscale` internally:
```cpp
void Painter::blit_scaled(const Rect& dst_rect, const GraphicsBitmap& source, const Rect& src_rect)
```
The rest of the code doesn't have to change, just the API.